python/modules/Stop0lObjectsProducer.py

```diff
@@ -133,21 +133,23 @@ def SelSoftb(self, isv):
 
 
     def SelJets(self, jet):
-        if jet.pt < 20 or math.fabs(jet.eta) > 2.4 :
+        if jet.pt < 20 or math.fabs(jet.eta) > 2.4:
             return False
         return True
 
     def SelPhotons(self, photon):
-        if photon.pt < 200:
+        if photon.pt < 220:
             return False
         abeta = math.fabs(photon.eta)
-        if (abeta > 1.442 and abeta < 1.566) or (abeta > 2.5):
+        if (abeta > 1.4442 and abeta < 1.566) or (abeta > 2.5):
             return False
-        ## cut-base ID, 2^0 loose ID
-        cutbase =  photon.cutBasedBitmap  if self.era != "2016" else photon.cutBased
-        if not cutbase & 0b1:
-            return False
-        return True
+        # --- Photon ID: cut-based medium ID --- #
+        # 2016:      Use Photon_cutBased       : Int_t cut-based Spring16-V2p2 ID (0:fail, 1: :loose, 2:medium, 3:tight)
+        # 2017,2018: Use Photon_cutBasedBitmap : Int_t cut-based ID bitmap, 2^(0:loose, 1: medium, 2:tight); should be 2017 V2
+        if self.era == "2016":
+            return bool(photon.cutBased > 1) 
```

Comment:
Can you check whether these ID are inclusive? For Medium ID, cutBase == 2 or cutBase >= 2? Which one is correct? Also (cutBased> 1) returned a bool, no need for converting it

Reply:
As far as I know (in discussions with @pastika) for medium ID cutBase == 2 is not correct, and cutBase >= 2 is correct.

Reply:
Why? That depends on whether the cuts are inclusive or not.

Reply:
Note that for integer IDs, the ID can only have one value (say 2 or 3) as it is not bitwise. That means tight photons are 3... but we should also count them as medium photons.

Reply:
```
146         # --- Photon ID: cut-based medium ID --- #
147         # 2016:      Use Photon_cutBased       : Int_t cut-based Spring16-V2p2 ID (0:fail, 1: :loose, 2:medium, 3:tight)
148         # 2017,2018: Use Photon_cutBasedBitmap : Int_t cut-based ID bitmap, 2^(0:loose, 1: medium, 2:tight); should be 2017 V2
```

Reply:
No, if the definition of medium ID is == 2, then we want to use == 2 , as the SF will be corresponding to that. >=2 will include both medium and tight ID. If they are inclusive (tight is a subset of medium), then we are OK. If not, we need to be exact. You can do a quick check for this

Reply:
I don't know if we can test that tight photons are medium (as that info is not kept... the value is only 2 or 3).

Reply:
To check we would have to have the ID definitions right?

Reply:
I guess we have to ask someone who knows...

Reply:
Ok, are we ok with this ? Any follow up on this ?

Reply:
@zhenbinwu yes, this is OK

https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/python/photons_cff.py#L228

look here for more details about the way photon ID is stored in 2016

Its not totally clear what the 2016 recommendation actually is, but I think it is 
https://twiki.cern.ch/twiki/bin/view/CMS/CutBasedPhotonIdentificationRun2#Offline_selection_criteria_AN2

Reply:
ok